beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go

```diff
@@ -52,19 +54,148 @@ type eth1DataAggregatedVote struct {
 	votes int
 }
 
-// GetBlockBeacon is called by a proposer during its assigned slot to request a block to sign
-// by passing in the slot and the signed randao reveal of the slot.
-// Once implemented, this method will DEPRECATE GetBlock.
+// blockData required to create a beacon block.
+type blockData struct {
+	ParentRoot        []byte
+	Graffiti          [32]byte
+	ProposerIdx       types.ValidatorIndex
+	Eth1Data          *ethpb.Eth1Data
+	Deposits          []*ethpb.Deposit
+	Attestations      []*ethpb.Attestation
+	ProposerSlashings []*ethpb.ProposerSlashing
+	AttesterSlashings []*ethpb.AttesterSlashing
+	VoluntaryExits    []*ethpb.SignedVoluntaryExit
+}
+
+// GetBeaconBlock is called by a proposer during its assigned slot to request a block to sign
+// by passing in the slot and the signed randao reveal of the slot. Returns phase0 beacon blocks
+// before the Altair fork epoch and Altair blocks post-fork epoch.
 func (vs *Server) GetBeaconBlock(ctx context.Context, req *ethpb.BlockRequest) (*ethpb.GenericBeaconBlock, error) {
-	return nil, status.Error(codes.Unimplemented, "Unimplemented")
+	ctx, span := trace.StartSpan(ctx, "ProposerServer.GetBlockAltair")
+	defer span.End()
+	span.AddAttributes(trace.Int64Attribute("slot", int64(req.Slot)))
+	if core.SlotToEpoch(req.Slot) < params.BeaconConfig().AltairForkEpoch {
+		blk, err := vs.getPhase0BeaconBlock(ctx, req)
+		if err != nil {
+			return nil, err
```

Comment:
Seems odd that this does not return grpc error status codes but the helper `getPhase0BeaconBlock` returns them instead. Don't feel strongly but just wanted to point this out

Reply:
Yep, amending